src/gsy_e/gsy_e_core/simulation/simulation.py

```diff
@@ -152,6 +152,8 @@ def run(self, initial_slot: int = 0) -> None:
             if self._setup.started_from_cli:
                 self._run_cli_execute_cycle(initial_slot, tick_resume)
             else:
+                # update status of the simulation before executing it.
+                self._results.update_and_send_results(simulation=self)
```

Comment:
This is the actual fix, right?

Reply:
Yes. This will fix the `running` status for SCM and non-SCM simulations.

Reply:
Wait, why is this the actual fix? Isn't the actual fix the one on this line? https://github.com/gridsingularity/gsy-e/pull/1542/files#diff-dd05e0495c9ab88ead3805c4dc846ef58cd1d882f667c952e12e721812338f2aR158
Or maybe I didn't get exactly what the GSYE-388 ticket is asking?

Regarding this specific line. My first question would have been "If the simulation hasn't started yet, why do we need to update the status?".
AFAIU, this fix only applies to simulations that were stopped and that we're now restarting. Is this correct?

Reply:
It depends on how we define start of a simulation. That line of code is running just before the real start of simulation loop.

Reply:
And for the second question, you're right. It only runs once and does not run in case of a paused simulation.